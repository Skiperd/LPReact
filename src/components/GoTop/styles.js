import styled,{css} from 'styled-components';
import {theme } from '../../styles/theme'

export const Container = styled.a`
  ${({theme}) => css`
    position: fixed;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    z-index: 6;
    /* opacity: 0.5; */
    cursor: pointer;
    border-radius: 10px;
  `}
`;
