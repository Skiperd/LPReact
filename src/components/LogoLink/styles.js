import styled,{css} from 'styled-components';
import {theme } from '../../styles/theme'
export const Container = styled.a`
  ${({theme}) => css`
    text-decoration: none;
    align-items: center;
    color: inherit;

    > img {
      max-height: 3rem;
    }
  `}
`;
