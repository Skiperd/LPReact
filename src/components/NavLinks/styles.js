import styled,{css} from 'styled-components';
import { theme } from '../../styles/theme'
export const Container = styled.nav`
  ${({theme}) => css`
    display: flex;
    flex-flow: row wrap;

    @media ${theme.media.lteMedium} {
      flex-flow: column wrap;
      align-content: center;
    }
  `}
`;
