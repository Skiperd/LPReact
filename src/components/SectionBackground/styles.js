import styled,{css} from 'styled-components';
import {theme } from '../../styles/theme'

const containerBackgroundActivate = () => css`
  background: ${theme.colors.primaryColor};
  color: ${theme.colors.white};
`

export const Container = styled.div`
  ${({theme, background}) => css`
    background: ${theme.colors.white};
    color: ${theme.colors.primaryColor};
    ${background && containerBackgroundActivate(theme)};
    min-height: 100vh;
    display: flex;
    align-items: center;
  `}
`;

