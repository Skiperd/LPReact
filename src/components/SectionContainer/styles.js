import styled,{css} from 'styled-components';
import {theme } from '../../styles/theme'
export const Container = styled.div`
  ${({theme}) => css`
    max-width: 1200px;
    margin: 0 auto;
    padding: ${theme.spacings.large}
  `}
`;
