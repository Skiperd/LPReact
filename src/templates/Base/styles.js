import styled,{css} from 'styled-components';
import {theme } from '../../styles/theme'
import {Container as TextComponent} from '../../components/TextComponent/styles'

export const Container = styled.div`
  ${({theme}) => css`
  /* padding-top: 5.4rem;

  @media ${theme.media.lteMedium} {
    padding-top: 0;
  } */
  `}
`;
