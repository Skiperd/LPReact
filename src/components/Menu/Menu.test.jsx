import { screen } from "@testing-library/react";
import { renderTheme } from "../../styles/render-theme";
import { Menu } from ".";
import linksMock from "../NavLinks/mock";
const logoData = {
  text: "Logo",
  link: "#target",
};

describe("<Menu />", () => {
  it("should render Logo and Main Menu Nav", () => {
    const { container } = renderTheme(
      <Menu links={linksMock} logoData={logoData}>
        Children
      </Menu>
    );
    expect(screen.getByRole("heading", { name: "Logo" })).toBeInTheDocument();
    expect(
      screen.getByRole("navigation", { name: "Main menu" })
    ).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("should render menu mobile and button for open and close the menu", () => {
    const { container } = renderTheme(
      <Menu links={linksMock} logoData={logoData}>
        Children
      </Menu>
    );

    const button = screen.getByLabelText("Open/Close menu");
    const menuContainer = button.nextSibling;

    // expect(button).toHaveStyleRule("display", "none");
  });
});
