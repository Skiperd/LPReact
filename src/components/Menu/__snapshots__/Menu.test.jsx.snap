// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should render Logo and Main Menu Nav 1`] = `
<div>
  <button
    aria-label="Open/Close menu"
    class="sc-eCYdKt fdxqoS"
  >
    <svg
      aria-hidden="true"
      aria-label="Open menu"
      class="StyledIconBase-sc-ea9ulj-0 llsAVB"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
  <div
    class="sc-dkzCjb bEueyH"
  >
    <div
      class="sc-bczSft cptumk"
    >
      <div
        class="sc-hKMtDE kwLQLU"
      >
        <h1
          class="sc-gsnScm iKcdGN"
        >
          <a
            class="sc-jSMdHm gbpaZR"
            href="#target"
          >
            <span>
              Logo
            </span>
          </a>
        </h1>
        <nav
          aria-label="Main menu"
          class="sc-gKXNOb eXENYG"
        >
          <a
            class="sc-iBkiHk cujlfa"
            href="#Target 1"
            target="_self"
          >
            Link 1
          </a>
          <a
            class="sc-iBkiHk cujlfa"
            href="#Target 2"
            target="_self"
          >
            Link 2
          </a>
          <a
            class="sc-iBkiHk cujlfa"
            href="#Target 3"
            target="_self"
          >
            Link 3
          </a>
          <a
            class="sc-iBkiHk cujlfa"
            href="#Target 4"
            target="_self"
          >
            Link 4
          </a>
          <a
            class="sc-iBkiHk cujlfa"
            href="#Target 5"
            target="_self"
          >
            Link 5
          </a>
          <a
            class="sc-iBkiHk cujlfa"
            href="#Target 6"
            target="_self"
          >
            Link 6
          </a>
          <a
            class="sc-iBkiHk cujlfa"
            href="#Target 7"
            target="_self"
          >
            Link 7
          </a>
          <a
            class="sc-iBkiHk cujlfa"
            href="#Target 8"
            target="_self"
          >
            Link 8
          </a>
          <a
            class="sc-iBkiHk cujlfa"
            href="#Target 9"
            target="_self"
          >
            Link 9
          </a>
          <a
            class="sc-iBkiHk cujlfa"
            href="#Target 10"
            target="_self"
          >
            Link 10
          </a>
        </nav>
      </div>
    </div>
  </div>
</div>
`;
